### PR TITLE
Simplify parts of the IC handler, and fix some bugs

### LINKED ISSUE #226: [Bug 2.8] Checking "Pre" on an emote with modifier 5 (zoom) causes infinite IC hang until interrupted by a shout
Via Door#0513:

> So it's been a long ass time since i've done ini editing, so I can't tell you why this is happening. But for some reason, some of the zooms, the ones that have the .ini written "Zoom#-#/zoom#5#", if you pre up a zoom, the message won't display, and someone else has to do a shout to interupt it. Characters that are written "Zoom#-#/zoom#6#" are able to use a pre on their zoom just fine. Idk if this is intentional or not. But seeing as everyone's ini is like that, if they haven't edited it. Felt like it was worth a mention

Confirmed via my own testing to occur on multiple servers.

### PR DESCRIPTION
**Pull request by scatterflower**
Fixes #226 and #169

### CODE REVIEW

File: src/courtroom.cpp [hunk 1]
```diff
@@ -1788,89 +1813,51 @@ void Courtroom::handle_chatmessage(QStringList *p_contents)
   }
 
   int f_char_id = m_chatmessage[CHAR_ID].toInt();
+  const bool is_spectator = (f_char_id == -1);
 
-  if (f_char_id >= 0 && f_char_id >= char_list.size())
+  if (f_char_id >= char_list.size())
```
Comment (stonedDiscord) [inline src/courtroom.cpp:1818]: what if it's -3

Comment (scatterflower) [reply src/courtroom.cpp:1818]: -3 will never be >= char_list.size() so it doesn't matter

Comment (scatterflower) [reply src/courtroom.cpp]: fixed to include negative check, which didn't work in the first place

**Comment by stonedDiscord** [comment]
epic gamer moment

**Pull request merged**